geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java

```diff
@@ -1838,6 +1838,9 @@ protected void reinitialize(Socket socket, ClientProxyMembershipID proxyId, Cach
     this._messageDispatcher._messageQueue.setPrimary(ip);
     this._messageDispatcher._messageQueue.setClientConflation(cc);
 
+    // Reset the _socketClosed AtomicBoolean
+    this._socketClosed.compareAndSet(true, false);
```

Comment:
It seems rather suspicious to compare and set but on do anything with the failure case. You could just `set` to `false`.